### ISSUE TITLE
pslist Page Fault

### ISSUE THREAD

**Issue by siftuser** [volatilityfoundation/volatility3#108]
`python3 vol.py -f memory.dmp windows.pslist` throws following error after showing few processes ... any clue ? thanks

```
0	0		0xad0ceefb9080	0	-	N/A	False	N/A	N/ATraceback (most recent call last):
  File "vol.py", line 10, in <module>
    volatility.cli.main()
  File "/volatility3/volatility/cli/__init__.py", line 442, in main
    CommandLine().run()
  File "/volatility3/volatility/cli/__init__.py", line 269, in run
    renderers[args.renderer]().render(constructed.run())
  File "/volatility3/volatility/cli/text_renderer.py", line 159, in render
    grid.populate(visitor, outfd)
  File "/volatility3/volatility/framework/renderers/__init__.py", line 196, in populate
    for (level, item) in self._generator:
  File "/volatility3/volatility/framework/plugins/windows/pslist.py", line 137, in _generator
    yield (0, (proc.UniqueProcessId, proc.InheritedFromUniqueProcessId,
  File "/volatility3/volatility/framework/objects/__init__.py", line 681, in __getattr__
    native_layer_name = self.vol.native_layer_name))
  File "/volatility3/volatility/framework/objects/templates.py", line 72, in __call__
    return self.vol.object_class(context = context, object_info = object_info, **arguments)
  File "/volatility3/volatility/framework/objects/__init__.py", line 113, in __new__
    value = cls._unmarshall(context, data_format, object_info)
  File "/volatility3/volatility/framework/objects/__init__.py", line 284, in _unmarshall
    data = context.layers.read(object_info.layer_name, object_info.offset, length)
  File "/volatility3/volatility/framework/interfaces/layers.py", line 492, in read
    return self[layer].read(offset, length, pad)
  File "/volatility3/volatility/framework/layers/linear.py", line 37, in read
    for (offset, mapped_offset, mapped_length, layer) in self.mapping(offset, length, ignore_errors = pad):
  File "/volatility3/volatility/framework/layers/intel.py", line 195, in mapping
    chunk_offset, page_size, layer_name = self._translate(offset)
  File "/volatility3/volatility/framework/layers/intel.py", line 317, in _translate
    return self._translate_swap(self, offset, self._bits_per_register // 2)
  File "/volatility3/volatility/framework/layers/intel.py", line 275, in _translate_swap
    return super()._translate(offset)
  File "/volatility3/volatility/framework/layers/intel.py", line 100, in _translate
    entry, position = self._translate_entry(offset)
  File "/volatility3/volatility/framework/layers/intel.py", line 143, in _translate_entry
    "Page Fault at entry " + hex(entry) + " in table " + name)
volatility.framework.exceptions.PagedInvalidAddressException: Page Fault at entry 0xb9b76063 in table page table

```

**Comment by ikelos** [comment]
This suggests that the image has some issues, specifically that a lookup for a memory address carried out in order to list the processes, wasn't classed as valid within the memory.  In volatility 2 we used to simply ignore errors like this and carry on blindly, but the user would never know whether we'd done that or not, so for volatility 3 we made a conscious decision to not try to mask over errors, but make sure the user was aware of them whenever it happened.

It could be this is smear from the acquisition tool taking time to image the memory (where a reference became invalid further down the line), or it may just be a broken image.  A third highly unlikely option is that it might have found a kernel header from another kernel image in memory somehow, and so based its table and symbols off that, but I don't think that's a realistic possibility, I'd have expected it to break somewhere sooner if that could've happened.

If you're able to re-image the box I'd recommend it, otherwise you can try `psscan` which will look for everything within memory that looks like it could be a process.  It's more likely to give you bad results, but it _shouldn't_ complain about invalid addresses in any way.  I'd also be interested to know what results (if there are more or not) from using volatility 2 on the same sample?

I'm not sure what more we can do to offer help, but I'll leave the issue open for a few days in case either of us gets something else to add to it...  5:)

**Comment by siftuser** [comment]
@ikelos many thanks for such a comprehensive explanation! Will try Vol2 & let you know if thats different. As you suggested .. psscan runs on Will also try to reimage the mem. 

FWIW - I used winpmem & dumptit for this collection ... winpmem collection is supporting  more plugins than dumpit so far

**Comment by atcuno** [comment]
@siftuser - could you please confirm that you are on the latest git checkout of master?

**Comment by siftuser** [comment]
thats correct

**Comment by siftuser** [comment]
> I'd also be interested to know what results (if there are more or not) from using volatility 2 on the same sample?
> 

@ikelos , FYI - vol2 works like a charm on same image file

**Comment by ikelos** [comment]
Hmmm, interesting, thanks for letting us know!  Is the memory sample one you'd be willing to share with us so we can try to diagnose the problem?  Not to worry if not, it'd just speed up finding a solution...  5:)

**Comment by ikelos** [comment]
Also, just for clarity, when you say works like a charm, could to include the output you get (for both ideally) so we can tell if it's returning more results, or just ignoring some errors?  Snippets make it difficult to tell the quality and quantity that each program returns...  5;)

**Comment by ikelos** [comment]
Just as a note, this issue should be improved by #124 (in that the exception that occurred will provide possible reasons and remedies).  We're unlikely to be able to fix it without getting a memory image that exhibits the problem and testing it.  Any word on whether the memory sample would be available for us to try and figure out what's going wrong?

**Comment by ikelos** [comment]
Having merged #124, I'm going to close this.  Feel free to reopen it if you feel there's more we can do to solve the issue...  5:)
